src/Symfony/Component/HttpFoundation/RedirectResponse.php

```diff
@@ -41,6 +41,10 @@ public function __construct($url, $status = 302, $headers = array())
         if (!$this->isRedirect()) {
             throw new \InvalidArgumentException(sprintf('The HTTP status code is not a redirect ("%s" given).', $status));
         }
+
+        if (301 == $status && !array_key_exists('cache-control', $headers)) {
```

Comment:
@fabpot, in Response class headers already computed and contains unnecessary 'no-cache' directive. We can't check wether is was set automatically or explicitly by user. see `\Symfony\Component\HttpFoundation\ResponseHeaderBag::computeCacheControlValue`